crates/toml/src/ser.rs

```diff
@@ -91,6 +92,60 @@ where
     Ok(dst)
 }
 
+/// Serialize the given data structure as TOML into the IO stream.
+///
+/// Serialization can fail if `T`'s implementation of `Serialize` decides to
+/// fail, if `T` contains a map with non-string keys, or if `T` attempts to
+/// serialize an unsupported datatype such as an enum, tuple, or tuple struct.
+///
+/// # Examples
+///
+/// ```
+/// #[macro_use]
+/// extern crate serde_derive;
+/// extern crate toml;
+///
+/// use std::fs::File;
+///
+/// #[derive(Serialize)]
+/// struct Config {
+///     database: Database,
+/// }
+///
+/// #[derive(Serialize)]
+/// struct Database {
+///     ip: String,
+///     port: Vec<u16>,
+///     connection_max: u32,
+///     enabled: bool,
+/// }
+///
+/// fn main() {
+/// # }
+/// # fn fake_main() {
+///     let config = Config {
+///         database: Database {
+///             ip: "192.168.1.1".to_string(),
+///             port: vec![8001, 8002, 8003],
+///             connection_max: 5000,
+///             enabled: false,
+///         },
+///     };
+///
+///     let mut file = File::create("config.toml").unwrap();
+///     toml::to_writer(&mut file, &config).unwrap();
+/// }
+/// ```
+pub fn to_writer<W, T: ?Sized>(writer: &mut W, value: &T) -> Result<(), Error>
```

Comment:
I see that `serde_json` offers a `to_writer_pretty`, should we do the same?

Reply:
There's no minified toml, is there? It's always "pretty".

Reply:
There are different ways to format the output.  See https://docs.rs/toml/latest/toml/ser/struct.Serializer.html#method.pretty